Core/Utilities/ElasticSearchDB.py

```diff
@@ -0,0 +1,272 @@
+  def query( self, index, query ):
+    """It exexutes a query and it returns the result
+    query is a dictionary. More info: search for elasticsearch dsl
+    
+    :param self: self reference
+    :param dict query: It is the query in ElasticSerach DSL language
+     
+    """
+    return self.__client.search( index = index, body = query )
+  
+  def _Search( self, indexname ):
+    """
+    it returns the object which can be used for reatriving ceratin value from the DB
+    """
+    return  Search( using = self.__client, index = indexname )
+  
+  ########################################################################
+  def _Q( self, name_or_query = 'match', **params ):
+    """
+    It is a wrapper to ElasticDSL Query module used to create a query object. 
+    :param name_or_query str is the type of the query
+    """
+    return Q( name_or_query, **params )
+  
+  def _A( self, name_or_agg, aggsfilter = None, **params ):
+    """
+    It us a wrapper to ElasticDSL aggregation module, used to create an aggregation
+    """
+    return A( name_or_agg, aggsfilter, **params )
+  ########################################################################
+  def __tryToConnect( self ):
+    """Before we use the database we try to connect and retrive the cluster name
+    
+    :param self: self reference
+         
+    """
+    try:
+      if self.__client.ping():
+        result = self.__client.info()
+        self.setClusterName ( result.get( "cluster_name", " " ) )
+        self.log.info( "Database info", result )
+        self._connected = True
+      else:
+        self.log.error( "Cannot connect to the database!" )
+    except ConnectionError as e:
+      self.log.error( e )
+      self._connected = False 
+
+  ########################################################################
+  def getIndexes( self ):
+    """
+    It returns the available indexes...
+    """
+    return [ index for index in self.__client.indices.get_aliases() ]
+  
+  ########################################################################
+  def getDocTypes( self, indexName ):
+    """
+    :param str indexName is the number of the index...
+    :return S_OK or S_ERROR 
+    """
+    try:
+      result = self.__client.indices.get_mapping( indexName )
+    except Exception as e:
+      gLogger.error( e )
+    doctype = ''
+    for i in result:
+      if len( result[i].get( 'mappings', {} ) ) == 0:
+        return S_ERROR( "%s does not exists!" % indexName )
+      doctype = result[i]['mappings']
+      break
+    return S_OK( doctype ) 
+  
+  ########################################################################
+  def isExists( self, indexName ):
+    """
+    it checks the existance of an index
+    :param str indexName: the name of the index
+    """
+    return self.__client.indices.exists( indexName )
+  
+  ########################################################################
+  def generateFullIndexName( self, indexName ):
```

Comment:
either it should return a standard dirac structure, or it should be private/protected method